src/common/procedure/src/local/runner.rs

```diff
@@ -98,10 +93,9 @@ impl Runner {
             self.manager_ctx.notify_by_subprocedure(parent_id);
         }
 
-        if let Some(key) = &lock_key {
-            self.manager_ctx
-                .lock_map
-                .release_lock(key.key(), self.meta.id);
+        // Release lock in reverse order.
+        for key in self.meta.lock_key.keys().iter().rev() {
```

Comment:
Locking and unlocking need to be in reverse order, so perhaps two api's can be provided: keys_to_lock() , keys_to_unlock()